install_latest_blue_onyx.ps1

```diff
@@ -164,59 +182,121 @@ try {
 
     # Cleanup the temp_unzip folder
     Remove-Item -Path $tempExtractPath -Recurse -Force
 
     # --- 8. Add that folder to the PATH (for the user environment) ---
-    Write-Host "Adding $destinationPath to User PATH..."
-    $userPath = [System.Environment]::GetEnvironmentVariable("PATH", "User")
-    
+    Write-Host "Adding $destinationPath to User PATH..." -ForegroundColor Green
+    $userPath = [System.Environment]::GetEnvironmentVariable("PATH", "Machine")
+
     if ($userPath -notlike "*$destinationPath*") {
         if ([string]::IsNullOrEmpty($userPath)) {
             $newPath = $destinationPath
         } else {
             $newPath = "$userPath;$destinationPath"
         }
-        
-        [System.Environment]::SetEnvironmentVariable("PATH", $newPath, "User")
+
+        [System.Environment]::SetEnvironmentVariable("PATH", $newPath, "Machine")
         # Also update the current session PATH so user can test immediately
         $env:PATH = "$($env:PATH);$destinationPath"
     } else {
         Write-Host "Path already contains $destinationPath, skipping update."
     }
 
     # --- 9. Run blue_onyx.exe to download all models ---
-    Write-Host "Downloading all models into $destinationPath..."
+    Write-Host "Downloading all models into $destinationPath..." -ForegroundColor DarkMagenta
     $exePath = Join-Path $destinationPath "blue_onyx.exe"
     if (-not (Test-Path $exePath)) {
         throw "Could not find blue_onyx.exe at $exePath. Installation may have failed or the file may be missing."
     }
-    # We'll run the exe in a separate process. 
+    # We'll run the exe in a separate process.
     # This will download models into .blue-onyx folder.
     & $exePath --download-model-path $destinationPath
 
+    # Now that we can see all the GPU's identifed when downloading all models.
+    # Select which GPU blue_onyx will use in bat files. 
+    # Import necessary namespace
+    Add-Type -AssemblyName System.Windows.Forms
+
+    # Get a list of available GPUs (replace with your actual GPU detection method)
+	# We want to exclude RDP Displays, and Sort the list like how it is in Task Manager so the Index is correct. 
+	$gpus = Get-CimInstance Win32_VideoController | Select-Object Name | Where-Object name -NotMatch "Microsoft Remote Display Adapter" |Sort-Object Name
+    $gpuNames = $gpus.Name
+
+    # Create a form
+    $form = New-Object System.Windows.Forms.Form
+    $form.Text = "Blue Onyx GPU Selection"
+    $form.Size = New-Object System.Drawing.Size(350, 210)
+    $form.StartPosition = 'CenterScreen'
+
+    # Create a label
+    $label = New-Object System.Windows.Forms.Label
+    $label.Text = "Select GPU:"
+    $label.Location = New-Object System.Drawing.Point(20, 20)
+    $form.Controls.Add($label)
+
+    # Create a combobox
+    $comboBox = New-Object System.Windows.Forms.ComboBox
+    $comboBox.Location = New-Object System.Drawing.Point(20, 50)
+    $comboBox.Width = 300
+    $comboBox.DataSource = $gpuNames
```

Comment:
I think this one fails on my machine, here is my log:

```bash
Downloading version.json from https://github.com/xnorpx/blue-onyx/releases/latest/download/version.json to C:\Users\xnorp\AppData\Local\Temp\BlueOnyxInstall\version.json...
Parsing version.json...
Version: 0.3.0
ZIP URL: https://github.com/xnorpx/blue-onyx/releases/latest/download/blue_onyx-0.3.0-win.zip
SHA256 URL: https://github.com/xnorpx/blue-onyx/releases/latest/download/blue_onyx-0.3.0-win.zip.sha256
Found cached ZIP file at C:\Users\xnorp\AppData\Local\Temp\BlueOnyxInstall\blue_onyx-0.3.0-win.zip. Skipping download...
Found cached SHA256 file at C:\Users\xnorp\AppData\Local\Temp\BlueOnyxInstall\blue_onyx-0.3.0-win.zip.sha256. Skipping download...
Verifying ZIP file integrity...
Expected SHA256: 48945fe7808d315794d34b16b6f097c67a666b34d5b7568e08c2dce28761f0a4
Actual   SHA256: 48945fe7808d315794d34b16b6f097c67a666b34d5b7568e08c2dce28761f0a4
SHA256 verification successful.
Extracting ZIP contents to C:\Users\xnorp\.blue-onyx\temp_unzip...
Single top-level folder found, flattening from: C:\Users\xnorp\.blue-onyx\temp_unzip\blue_onyx-0.3.0-win
Checking if blue_onyx.exe is running...
blue_onyx.exe is not running. Proceeding with installation...
Overwriting existing files in C:\Users\xnorp\.blue-onyx with the new files...
Adding C:\Users\xnorp\.blue-onyx to User PATH...
Downloading all models into C:\Users\xnorp\.blue-onyx...
2024-12-27T05:15:51.447979Z  INFO blue_onyx: Logging initialized log_level=Info
2024-12-27T05:15:51.449629Z  INFO blue_onyx::system_info: System Information:
2024-12-27T05:15:51.450029Z  INFO blue_onyx::system_info: CPU | GenuineIntel | 12th Gen Intel(R) Core(TM) i7-12700F | 12 Cores | 20 Logical Cores
2024-12-27T05:15:51.452470Z  INFO blue_onyx::system_info: GPU 0 | NVIDIA GeForce RTX 4080 SUPER | Vendor: 10DE, Device: 2702, Dedicated VRAM: 16047.00 MB
2024-12-27T05:15:51.452698Z  INFO blue_onyx::system_info: GPU 1 | NVIDIA GeForce RTX 4080 SUPER | Vendor: 10DE, Device: 2702, Dedicated VRAM: 16047.00 MB
2024-12-27T05:15:51.454525Z  INFO hf_hub: Token file not found "C:\\Users\\xnorp\\.cache\\huggingface\\token"
2024-12-27T05:15:51.487368Z  INFO blue_onyx::download_models: Copied rt-detrv2-s.onnx to C:\Users\xnorp\.blue-onyx\rt-detrv2-s.onnx
2024-12-27T05:15:51.576470Z  INFO blue_onyx::download_models: Copied rt-detrv2-ms.onnx to C:\Users\xnorp\.blue-onyx\rt-detrv2-ms.onnx
2024-12-27T05:15:51.633026Z  INFO blue_onyx::download_models: Copied rt-detrv2-m.onnx to C:\Users\xnorp\.blue-onyx\rt-detrv2-m.onnx
2024-12-27T05:15:51.713769Z  INFO blue_onyx::download_models: Copied rt-detrv2-l.onnx to C:\Users\xnorp\.blue-onyx\rt-detrv2-l.onnx
2024-12-27T05:15:51.909570Z  INFO blue_onyx::download_models: Copied rt-detrv2-x.onnx to C:\Users\xnorp\.blue-onyx\rt-detrv2-x.onnx
2024-12-27T05:15:51.909711Z  INFO blue_onyx::download_models: Succesfully downloaded models: ["rt-detrv2-s.onnx", "rt-detrv2-ms.onnx", "rt-detrv2-m.onnx", "rt-detrv2-l.onnx", "rt-detrv2-x.onnx"]

[ERROR] Exception setting "DataSource": "Complex DataBinding accepts as a data source either an IList or an IListSource."

[ERROR] Installation failed. Exiting...
```

Reply:
thats interesting. Are you running as Admin? or your self?

Reply:
This was with Admin, without admin it fails earlier.

Reply:
Are you able to run this in a Powershell and get an output:
```powershell
Get-CimInstance Win32_VideoController | Select-Object Name | Where-Object name -NotMatch "Microsoft Remote Display Adapter" |Sort-Object Name
```

Reply:
Get-CimInstance Win32_VideoController | Select-Object Name | Where-Object name -NotMatch "Microsoft Remote Display Adapter" |Sort-Object Name

Name
----
NVIDIA GeForce RTX 4080 SUPER

Reply:
trying a new if else statement that might help if the result of that command is only 1.

Reply:
![image](https://github.com/user-attachments/assets/b9cf38b1-66e0-43be-b7d9-ee2040868b80)

Reply:
can you test the new script to see if that helped or not.

Reply:
$gpus = @(Get-CimInstance Win32_VideoController | Select-Object Name | Where-Object name -NotMatch "Microsoft Remote Display Adapter" | Sort-Object Name)

you need to ensure gpus in array

Reply:
if that works for you awesome!  that change still works for me!